lib/rules/curly.js

```diff
@@ -290,7 +290,9 @@ module.exports = {
                 }
             } else if (multiOrNest) {
                 if (hasBlock && body.body.length === 1 && isOneLiner(body.body[0])) {
-                    expected = false;
+                    const leadingComments = sourceCode.getComments(body.body[0]).leading;
+
+                    expected = !!(leadingComments && leadingComments.length);
```

Comment:
This check should be able to just be `expected = !!(leadingComments.length)` now, since `sourceCode.getComments()` will return an empty array if none are found